server/src/main/java/org/spine3/server/aggregate/AggregatePartRepository.java

```diff
@@ -63,15 +64,15 @@ protected AggregatePartRepository(BoundedContext boundedContext, Class<R> rootCl
     public A create(I id) {
         final Constructor<A> entityConstructor = getEntityConstructor();
         final AggregateRoot<I> root = createAggregateRoot(id, getBoundedContext(), rootClass);
-        final A result = createAggregatePart(entityConstructor, id, root);
+        final A result = createAggregatePart(entityConstructor, root);
         return result;
     }
 
     @SuppressWarnings("MethodDoesntCallSuperMethod") // The call of the super method is not needed.
     @Override
     protected Constructor<A> getEntityConstructor() {
         final Constructor<A> result =
-                Aggregates.getAggregatePartConstructor(getEntityClass(), rootClass, getIdClass());
+                Aggregates.getAggregatePartConstructor(getEntityClass(), rootClass);
```

Comment:
Can we get the root class as a generic parameter of the aggregate class or, more straightforward, the repository class? If we can, we won't need to accept it as a constructor parameter.